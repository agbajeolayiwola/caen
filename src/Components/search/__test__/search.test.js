import React from 'react';
import ReactDOM from 'react-dom'
import SearchComp from '../index'
import { isTSAnyKeyword } from '@babel/types';
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'

it('renders whole without crashing',()=>{
    const div = document.createElement("div")
    ReactDOM.render(<SearchComp></SearchComp>, div)
})
it('renders input field correctly', ()=>{
    const {getByTestId} = render(<SearchComp></SearchComp>)
    expect(getByTestId('inputBtn')).toHaveTextContent('')
})
it('matches snapshot',()=>{
    const tree = renderer.create(<SearchComp></SearchComp>)
    expect(tree).toMatchSnapshot();
})