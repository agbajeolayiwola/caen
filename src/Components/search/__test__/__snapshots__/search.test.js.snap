// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="search"
>
  <form
    className="search_Form"
  >
    <input
      autoFocus={true}
      className="form-control"
      data-testid="inputText"
      onChange={[Function]}
      placeholder="Search"
      type="text"
    />
    <br />
    <input
      className="search_Btn"
      data-testid="inputBtn"
      onClick={[Function]}
      type="submit"
      value="Search"
    />
  </form>
  <div>
    <div>
      <div
        className="result_header_div"
      >
        <ul
          className="result_Header_ul"
        >
          <li>
            Avatar
          </li>
          <li>
            Name
          </li>
          <li>
            Status
          </li>
        </ul>
      </div>
    </div>
    <ul
      className="paginationButton"
    >
      <li
        className="previous paginDisabed"
      >
        <a
          aria-disabled="true"
          aria-label="Previous page"
          className="previousBtn"
          onClick={[Function]}
          onKeyPress={[Function]}
          rel="prev"
          role="button"
          tabIndex="0"
        >
          Prev
        </a>
      </li>
      <li
        className="next"
      >
        <a
          aria-disabled="false"
          aria-label="Next page"
          className="nextBtn"
          onClick={[Function]}
          onKeyPress={[Function]}
          rel="next"
          role="button"
          tabIndex="0"
        >
          next
        </a>
      </li>
    </ul>
  </div>
</div>
`;
